### PR TITLE
Add rtmedia_activity_content_html filter

### PR DESCRIPTION
**Pull request by dharmin**
Add rtmedia_activity_content to filter the content of the activity before save and fix some phpcs.

It have 3 arguments.

$activity_content Concatination of $activity_text and $media_list.
$activity_text       HTML markup of activity text.
$media_list          HTML markup of media in ul.

It'll help to change the order of the content display(default order is `activity text` and then `media list` or add some markup or separater between the `activity text`  and `media list`.

### CODE REVIEW

File: app/main/controllers/activity/RTMediaActivity.php [hunk 1]
```diff
@@ -34,16 +34,21 @@ function __construct( $media, $privacy = 0, $activity_text = false ) {
 		$this->privacy       = $privacy;
 	}
 
+	/**
+	 * Create html of activity/comments using text and media.
+	 * 
+	 * @param string $type Type of activity (activity/comment).
+	 */
```
Comment (ravatparmar) [inline app/main/controllers/activity/RTMediaActivity.php]: Please write about return type, and can we add more description about this function.

File: app/main/controllers/activity/RTMediaActivity.php [hunk 2]
```diff
@@ -34,16 +34,21 @@ function __construct( $media, $privacy = 0, $activity_text = false ) {
...
-		$html .= '<div class="rtmedia-'.$type.'-container">';
+		$activity_container_start = '<div class="rtmedia-'. esc_attr( $type ) .'-container">';
+		$activity_container_end   = '</div>';
 
+		$activity_text = '';
 		if ( ! empty( $this->activity_text ) && '&nbsp;' !== $this->activity_text ) {
-			$html .= '<div class="rtmedia-'.$type.'-text"><span>';
-			$html .= $this->activity_text;
-			$html .= '</span></div>';
+			$activity_text .= '<div class="rtmedia-' . esc_attr( $type ) . '-text"><span>';
```
Comment (ravatparmar) [inline app/main/controllers/activity/RTMediaActivity.php]: As we are redefining this whole function can we use sprintf in all the html for more readability?

`$activity_text .= sprintf( '<div class="rtmedia-%s-text"><span>', esc_attr( $type ) );`

Comment (ravatparmar) [reply app/main/controllers/activity/RTMediaActivity.php]: we can use this in one line. 

$activity_text .= sprintf( '<div class="rtmedia-%1$s-text"><span>%2$s</span></div>', esc_attr( $type ),  $this->activity_text );

check if require escaping in activity text.

File: app/main/controllers/activity/RTMediaActivity.php [hunk 3]
```diff
@@ -53,71 +58,72 @@ function create_activity_html( $type = 'activity' ) {
...
+				$media_content .= '<a href ="' . esc_url( get_rtmedia_permalink( $media->id ) ) . '">';
 			}
-			$li_content .= '<div class="rtmedia-item-thumbnail">';
+			$media_content .= '<div class="rtmedia-item-thumbnail">';
 
-			$li_content .= $this->media( $media );
+			$media_content .= $this->media( $media );
 
-			$li_content .= '</div>';
+			$media_content .= '</div>';
```
Comment (ravatparmar) [inline app/main/controllers/activity/RTMediaActivity.php]: we can use this in one line.

$media_content .= sprintf( '<div class="rtmedia-item-thumbnail">%s</div>', esc_html(  $this->media( $media ) ) );

check if require escaping in media.

File: app/main/controllers/activity/RTMediaActivity.php [hunk 4]
```diff
@@ -53,71 +58,72 @@ function create_activity_html( $type = 'activity' ) {
...
-				$li_content .= '<a href="' . esc_url( get_rtmedia_permalink( $media->id ) ) . '">';
+				$media_content .= '<a href="' . esc_url( get_rtmedia_permalink( $media->id ) ) . '">';
 			}
 
-			$li_content .= $media->media_title;
+			$media_content .= $media->media_title;
 			if ( 'photo' !== $media->media_type && 'document' !== $media->media_type && 'other' !== $media->media_type ) {
-				$li_content .= '</a>';
+				$media_content .= '</a>';
 			}
```
Comment (ravatparmar) [inline app/main/controllers/activity/RTMediaActivity.php]: We can also optimize these two if and we can put in if,else

File: app/main/controllers/activity/RTMediaActivity.php [hunk 5]
```diff
@@ -53,71 +68,110 @@ function create_activity_html( $type = 'activity' ) {
...
+					</div>
+					<div class="rtmedia-item-title">
+							<h4 title="%2$s">
+								%3$s
+							</h4>
+					</div>',
+					$this->media( $media ),
+					esc_attr( $media->media_title ),
+					$media->media_title
+				);
```
Comment (ravatparmar) [inline app/main/controllers/activity/RTMediaActivity.php]: More optimizable, but less customizable way.

```
$media_title = ( 'audio' === $media->media_type || 'video' === $media->media_type ) ? sprintf( '<a href="%s">%s</a>', esc_attr( $media->media_title ) ) : $media->media_title;

$media_inner_content =  sprintf(
                    '<div class="rtmedia-item-thumbnail">
                            %1$s
                    </div>
                    <div class="rtmedia-item-title">
                            <h4 title="%2$s">
                                %3$s
                            </h4>
                    </div>',
                    $this->media( $media ),
                    esc_attr( $media->media_title ),
                    $media_title
                );
if ( 'photo' === $media->media_type ) {
      $media_inner_content = sprintf( '<a href ="%1$s">%2$s</a>', $media_inner_content );
}

$media_content .= $media_inner_content
```

File: app/main/controllers/activity/RTMediaActivity.php [hunk 6]
```diff
@@ -53,71 +68,110 @@ function create_activity_html( $type = 'activity' ) {
...
-
-		return bp_activity_filter_kses( $html );
+		$media_container_end = '</ul>';
+
+		$media_list  = $media_container_start;
+		$media_list .= $media_content;
+		$media_list .= $media_container_end;
+
+		/**
+		 * Filters the output of the activity contents before save.
```
Comment (ravatparmar) [inline app/main/controllers/activity/RTMediaActivity.php]: before display

File: app/main/controllers/activity/RTMediaActivity.php [hunk 7]
```diff
@@ -53,71 +68,110 @@ function create_activity_html( $type = 'activity' ) {
...
+				esc_attr( $rtmedia_activity_ul_class ),
+				esc_attr( $count )
+			);
+		} else {
+			$media_container_start .= sprintf(
+				'<ul class="rtmedia-%1$s-list %2$s rtmedia-activity-media-length-%3$s">',
+				esc_attr( $type ),
+				esc_attr( $rtmedia_activity_ul_class ),
+				esc_attr( $count )
+			);
```
Comment (ravatparmar) [inline app/main/controllers/activity/RTMediaActivity.php]: Just a suggestion, we can optimize this also.

```$media_container_start_class = 'rtmedia-list',;
if ( 'activity' !== $type ) {
     $media_container_start_class = sprintf( 'rtmedia-%1$s-list', attrs );
}
$media_container_start .= sprintf( '<ul class="%1$s %2$s rtmedia-activity-media-length-%3$s">', attrs );```

File: app/main/controllers/activity/RTMediaActivity.php (hunk 5 above)
Comment (dharmin) [reply app/main/controllers/activity/RTMediaActivity.php]: I'm keeping it as it is, so it'll be easy to add modification in the future.

File: app/main/controllers/activity/RTMediaActivity.php (hunk 6 above)
Comment (dharmin) [reply app/main/controllers/activity/RTMediaActivity.php]: No, this function modifies the content before saving it into the database, so I've add `before save`.

File: app/main/controllers/activity/RTMediaActivity.php [hunk 8]
```diff
@@ -34,16 +34,31 @@ function __construct( $media, $privacy = 0, $activity_text = false ) {
 		$this->privacy       = $privacy;
 	}
 
-	function create_activity_html( $type = 'activity' ) {
-
-		$html = '';
+	/**
+	 * Function is used to generate HTML of activity/comments.
+	 * It combines the the rtMedia text and media in the activity content.
```
Comment (thrijith) [inline app/main/controllers/activity/RTMediaActivity.php]:
```suggestion
	 * It combines the rtMedia text and media in the activity content.
```
Minor typo

File: app/main/controllers/activity/RTMediaActivity.php [hunk 9]
```diff
@@ -53,71 +68,107 @@ function create_activity_html( $type = 'activity' ) {
...
+						<h4 title="%2$s">
+							<a href="%3$s">
+								%4$s
+							</a>
+						</h4>
+					</div>',
+					$this->media( $media ),
+					esc_attr( $media->media_title ),
+					esc_url( get_rtmedia_permalink( $media->id ) ),
+					$media->media_title
```
Comment (thrijith) [inline app/main/controllers/activity/RTMediaActivity.php]: This should probably be escaped.

File: app/main/controllers/activity/RTMediaActivity.php [hunk 10]
```diff
@@ -53,71 +68,107 @@ function create_activity_html( $type = 'activity' ) {
...
+					esc_url( get_rtmedia_permalink( $media->id ) ),
+					$this->media( $media ),
+					esc_attr( $media->media_title ),
+					$media->media_title
+				);
+			} elseif ( 'music' === $media->media_type || 'video' === $media->media_type ) {
+				// Markup for audio and video media type with link only on media (title).
+				$media_content .= sprintf(
+					'<div class="rtmedia-item-thumbnail">
+						%1$s
```
Comment (thrijith) [inline app/main/controllers/activity/RTMediaActivity.php]: Do we need numbered placeholders here? If we aren't re-using, `%s` should work.

File: app/main/controllers/activity/RTMediaActivity.php (hunk 5 above)
Comment (thrijith) [reply app/main/controllers/activity/RTMediaActivity.php]: Try to reuse the code here if possible, as if it's using same markup then change would be easy ( i.e at one place ), if a different type needs any specific change then we can implement as you have done now.

**Pull request merged**